### DISCUSSION TITLE
Report - (4th Sem.) Week 3 📢

### DISCUSSION THREAD

**Discussion by Nahlam4** [SE-TINF22B6/DHBWhub#562] [Weekly Report]
Greetings and welcome to our eleventh progress report! We're thrilled to update you on the latest news and developments since last week! 🥳

### **Software Architecture Documentation**

This Week, we expanded our SAD by an package diagram and an example sequence diagram for creating a comment. For a more detailed description of our software architecture, you can read our first draft of our [SAD document](https://github.com/SE-TINF22B6/DHBWhub/blob/master/docs/SOFTWARE_ARCHITECTURE_DOCUMENTATION.md). The first diagram depicts our architecture as an UML package diagram:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/f601faa3-2f42-4e05-9590-7d28ca062ea9)
  
On the next image, you can see a sequence diagram in the component view, for the action of commenting a post on the website:  
  
![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/efe36510-dbe7-4f9b-a1b5-b32777dc961a)

### **Developments on the frontend 🖥️**

This week, our team expanded last weeks Mockups for the friends and FAQ page to allow the usage of our platform by smartphone, providing easy access to our users.

<div style="display: flex; flex-direction: column;"> 
  <img src="https://github.com/SE-TINF22B6/DHBWhub/assets/111356450/c581e1ac-0290-4a3b-a884-bd29fa5307dc" alt="Friends - Smartphone View" /> 
  <img src="https://github.com/SE-TINF22B6/DHBWhub/assets/111356450/25e22fc3-3040-4ccc-b167-dd19abbe6250" alt="FAQ - Smartphone View" /> 
  <img src="https://github.com/SE-TINF22B6/DHBWhub/assets/111356450/cf79139c-f36a-4295-970a-e57258ef48a4" alt="Search - Smartphone View" /> 
</div>

Also, we created a Mockup for our notification Window: 

![Notification - PC View](https://github.com/SE-TINF22B6/DHBWhub/assets/111356450/4a856bea-152f-4851-8d1c-028085173447)
![Notification - Smartphone View](https://github.com/SE-TINF22B6/DHBWhub/assets/111356450/98496cd7-3e89-4a4d-a678-06453715e476)

We also finally implemented components for Events, Posts and Tags in the Frontend and startet implementing the functionality for our login/signup windows, so that we are once again one step closer to a functioning account-system. 

But we also havent slept on working on our Frontend, as it now already shows Popular Tags and upcoming Events:

![image](https://github.com/SE-TINF22B6/DHBWhub/assets/118728701/c174f0b5-6ac1-4bbd-9ac6-e5d109b9d5f0)

### **Developments on the backend 🧮**

 We further Improved security for our API and revised our models to align better with the Frontend, thus preparing for our next big goal: The correct and secure connection between our Front- and Backend. 

This week we also successfully deployed our Frontend to Digital Ocean  and our Website is now available at https://www.dhbwhub.de/

### **Main goals and priorities in the next weeks 📈**

In the upcoming weeks, we aim to complete the connection between frontend and backend so that our main functions like account interaction and the generation of content may begin. This includes the implementation of comments and the creation of Posts and Events. Also, the completion of mockups for the remaining components stays one of our top priorities, providing us clearer visualizations to guide the implementation of new features.

Sprint 7: What have we achieved ❓ What's to come❔
✅ Frontend: Implement 'Create post'-component
✅ Backend: Adding adjusted models, modelproposals and mapper
✅ Database: Insert example image into database
✅ Database+Backend: Create Tag-Table
✅ Set up Digital Ocean
✅ Backend: Add basic auth for endpoints
✅ Backend: Implement relevant friendship endpoints
✅ Create Mockup for FAQ page
✅ Create Mockup for Friends page
✅ Deploy Frontend to Digital Ocean
✅ Create Mobile Mockup for FAQ page
✅ Create Mobile Mockup for Friends page
✅ Create Mockup for Saved Posts
✅ Create Mockup for notifications window
✅ Create various React components
✅ Improve Backend for authentication (signup/login)
🚧 Adapt login & signup-dialog
🚧 Remove Herobrine

Until next week 💯

Your DHBWhub Team 🚀

**Comment by TomSarh** [comment]
Dear DHBWhub team,
The improvements you made to the frontend look really good and adding popular tags on the home page will be really useful for users to easily find subjects that they like. Great to see that you made other improvements to your API security, it's good seeing that you're making great and consistent progress on your project !

**Comment by sfrancisw** [comment]
Dear DHBWhub team,
those are some great updates! I really like your sprint list to overview what you've achieved and what you still have to do. It seems like you are well organized and ready to work on your next big steps! Keep up the good work.

Your team t2s

**Comment by derrrobin** [comment]
Dear DHBWhub team,

Fantastic updates! Your sprint list provides a clear overview of accomplishments and upcoming tasks, showcasing excellent teamwork and readiness for your next big steps. Keep up the great work! Your CookPit team
